web/src/engine/websites/Mangadon.ts

```diff
@@ -0,0 +1,133 @@
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return /https:\/\/mangadon\.me\/catalogs\/show\?tb=\d+$/.test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const mangaid = new URL(url).searchParams.get('tb');
+        const request = new Request(new URL(`/api/v1/comics/${mangaid}`, this.URI));
+        const { data } = await FetchJSON<APIResult>(request);
+        const title = (data as APIItem).attributes.name.trim();
+        return new Manga(this, provider, mangaid, title);
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const request = new Request(new URL(`/api/v1/comics`, this.URI));
+        const { data } = await FetchJSON<APIResult>(request);
+        return (data as APIItem[]).map(manga => new Manga(this, provider, manga.id, manga.attributes.name.trim()));
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const request = new Request(new URL(`/api/v1/comics/${manga.Identifier}?includes%5B%5D=episodes`, this.URI));
+        const { included } = await FetchJSON<APIResult>(request);
+        return included.map(chapter => new Chapter(this, manga, chapter.id, chapter.attributes.name.trim()));
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        const request = new Request(new URL(`/api/v1/episodes/${chapter.Identifier}?params%5Bcookie_signer%5D=true&params%5Bpages%5D=true`, this.URI));
+        const { data } = await FetchJSON<APIResult>(request);
+        const cookies = (data as APIItem).attributes.cookie_signer;
+
+        if (!cookies['CloudFront-Key-Pair-Id']) {
+            throw new Exception(R.Plugin_Common_Chapter_UnavailableError);
+        }
+        return (data as APIItem).attributes.pages.map(page => new Page(this, chapter, new URL(page.path, this.imgCDN), { ...cookies }));
+    }
+
+    public override async FetchImage(page: Page, priority: Priority, signal: AbortSignal): Promise<Blob> {
+        return this.imageTaskPool.Add(async () => {
+
+            //Compute cookie data for image request, otherwise we got a 403
+            const cookiesData = page.Parameters as CookieSigner;
+            const cookies: string[] = [];
+            Object.keys(cookiesData).forEach(name => {
+                const value = cookiesData[name];
+                cookies.push(`${name}=${value}`);
+            });
+
+            const request = new Request(page.Link, {
+                signal: signal,
+                headers: {
+                    Cookie: cookies.join(';'),
+                    Referer: page.Link.origin,
+                }
+            });
+
+            const response = await Fetch(request);
+            const blob = await response.blob();
```

Comment:
Move the descrambling outside of the task-pool by returning the promise with blob. No need to block the queue any longer after response was received so next image can be downloaded as soon as possible.